content/en/blog/_posts/2023-12-18-kubernetes-1-29-feature-loadbalancer-ip-mode-alpha.md

```diff
@@ -0,0 +1,102 @@
+exemplified in this blog by kube-proxy, 
+handle traffic from pods to the Service, within the cluster.
+
+## Background
+
+As of today, the different kube-proxy implementations(including ipvs and iptables) bind the external IP 
+of the service which type is set to LoadBalancer to each node. 
+This is achieved through iptables redirecting packets directly to the service and ipvs binding the IP 
+to one interface on the node. This feature is implemented for the following reasons:
+
+1. **Traffic path optimization:** Efficiently redirecting pod traffic sent to the load balancer IP 
+directly to the backend service by bypassing the load balancer.
+  
+2. **Handling load balancer packets:** Some load balancers send packets with the destination IP set to 
+the load balancer's IP. As a result, these packets need to be routed directly to the correct backend service 
+on the node to avoid loops.
+  
+## Problems
+
+However, there are several problems with the aforementioned behavior:
+
+1. **[Source IP](https://github.com/kubernetes/kubernetes/issues/79783):** 
+    Some cloud providers use the load balancer's IP as the source IP when 
+    transmitting packets to the node. In the ipvs mode of kube-proxy, 
+    there is a problem that health checks from the load balancer never return. This occurs because the reply packets 
+    would be forward to the local interface `kube-ipvs0`(where the load balancer's IP is bound to) 
+    and be subsequently ignored.
+  
+2. **[Feature loss at load balancer level](https://github.com/kubernetes/kubernetes/issues/66607):**
+    Certain cloud providers offer features(such as TLS termination, proxy protocol, etc.) at the
+    load balancer level.
+    Bypassing the load balancer results in the loss of these features when the packet reaches the service
+    (leading to protocol errors).
+  
+
+Currently, there is a [workaround](https://github.com/kubernetes/kubernetes/issues/66607#issuecomment-474513060) that involves setting the `hostname` of the `service.status.loabalancer.ingress` 
+to bypass kube-proxy binding. 
+But this is just a makeshift solution.
+
+## Solution
+
+In summary, providing an option for cloud providers to disable the current behavior would be highly beneficial.
+
+To address this, Kubernetes v1.29 introduces a new (alpha) `service.status.loadbalancer.ingress.ipMode` 
+field for a Service.
+This field specifies how the load balancer IP behaves and can be specified only when 
+the `.status.loadBalancer.ingress.ip` field is also specified.
+
+Two values are possible for `.status.loadBalancer.ingress.ipMode`: `"VIP"` and `"Proxy"`.
+The default value is "VIP", meaning that traffic delivered to the node 
+with the destination set to the load balancer's IP and port will be redirected to the backend service by kube-proxy.
+This preserves the existing behavior of kube-proxy. 
+The "Proxy" value is intended to prevent kube-proxy from binding the Load Balancer IP to the node in both ipvs and iptables modes. 
```

Comment:
```suggestion
The "Proxy" value is intended to prevent kube-proxy from binding the load balancer's IP address to the node
any mode (iptables, ipvs, nftables).
```
?

What about Windows behavior? There's another operating mode for kube-proxy and the article as written makes it sound like there isn't.

Reply:
Windows doesn't have nftables, but there is a `kube-proxy` mode for Windows: [`kernelspace`](https://kubernetes.io/docs/reference/networking/virtual-ips/#proxy-mode-kernelspace)

Reply:
Is there a k/k issue tracked?

Reply:
I create a new one(https://github.com/kubernetes/kubernetes/issues/122202).